pkg/controllers/provisioning/suite_test.go

```diff
@@ -554,19 +557,78 @@ var _ = Describe("Provisioning", func() {
 			Expect(*allocatable.Cpu()).To(Equal(resource.MustParse("4")))
 			Expect(*allocatable.Memory()).To(Equal(resource.MustParse("4Gi")))
 		})
+		It("should schedule based on the max resource requests of containers and initContainers with sidecar containers when initcontainer comes first", func() {
+			ExpectApplied(ctx, env.Client, test.NodePool())
+			pod := test.UnschedulablePod(test.PodOptions{
+				ResourceRequirements: v1.ResourceRequirements{
+					Limits:   v1.ResourceList{v1.ResourceCPU: resource.MustParse("5"), v1.ResourceMemory: resource.MustParse("1Gi")},
+					Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("5")},
+				},
+				InitContainers: []v1.Container{
+					{
+						Resources: v1.ResourceRequirements{
+							Limits:   v1.ResourceList{v1.ResourceCPU: resource.MustParse("4"), v1.ResourceMemory: resource.MustParse("2Gi")},
+							Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("1")},
+						},
+					},
+					{
+						RestartPolicy: lo.ToPtr(v1.ContainerRestartPolicyAlways),
+						Resources: v1.ResourceRequirements{
+							Limits:   v1.ResourceList{v1.ResourceCPU: resource.MustParse("5"), v1.ResourceMemory: resource.MustParse("3Gi")},
+							Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("5")},
+						},
+					},
+				},
+			})
+
+			ExpectProvisioned(ctx, env.Client, cluster, cloudProvider, prov, pod)
+			node := ExpectScheduled(ctx, env.Client, pod)
+			Expect(node.Status.Capacity.Cpu().Value()).To(BeNumerically(">", 10))
```

Comment:
Looking back at these test cases, I don't think that these checks ensure what they are supposed to ensure. Doing a Gt check here doesn't actually validate anything since just getting a big enough instance doesn't actually tell us anything about which set of resource requests are considered. The reason I know that is that (depending on which instance types are in our CloudProvider code), you might get the same instance type even if our logic was incorrect around how we were handling sidecar containers here. 

We should really do a check like in the previous test where we grab the instance type and then validate the resources that exist for that instance type are the same as what we would have expected to provision. Ideally, an equals operation here instead of a Gt check

Reply:
You can see the instance types that are leveraged here: https://github.com/kubernetes-sigs/karpenter/blob/main/pkg/cloudprovider/fake/cloudprovider.go#L164 and then override them with `cloudprovider.InstanceTypes` if you need more control for the specific test case